apps/next/features/banner/renderBanner.ts

```diff
@@ -0,0 +1,56 @@
+import { TwitchClientAuthService } from "@app/services/TwitchClientAuthService";
+import { remotionAxios, twitchAxios } from "@app/util/axios";
+import { getBannerEntry } from "@app/util/database/postgresHelpers";
+import type { Prisma } from "@prisma/client";
+import type { AxiosResponse } from "axios";
+
+export type TemplateRequestBody = {
+    foregroundId: string;
+    backgroundId: string;
+    foregroundProps: Record<string, unknown>;
+    backgroundProps: Record<string, unknown>;
+};
+
+/**
+ * @param userId
+ * @param twitchUserId
+ * @returns base64 of rendered banner
+ */
+export const renderBanner = async (userId: string, twitchUserId: string): Promise<string> => {
+    // get the banner info saved in Banner table
+    const bannerEntry = await getBannerEntry(userId);
+    if (bannerEntry === null) {
+        throw new Error('Could not find banner entry for user');
+    }
+
+    const authedTwitchAxios = await TwitchClientAuthService.authAxios(twitchAxios);
+
+    // get twitch stream info for user
+    // https://dev.twitch.tv/docs/api/reference#get-streams
+    const streamResponse = await authedTwitchAxios.get(`/helix/streams?user_id=${twitchUserId}`);
```

Comment:
if this is empty, they are not live. We should just void the update in this case.